### ISSUE TITLE
Low mAP score from training custom dataset

### ISSUE THREAD

**Issue by emiltan97** [dbolya/yolact#504]
Hi I am currently working on instance segmentation on food. I have followed the instructions repo wiki and started training the model with my custom dataset. The configuration is as follow : 

- 2 classes (food in general, and background)
- 9654 train images and 2439 validation images 
- backbone : yolact_resnet50

Below is the result : 
![result](https://user-images.githubusercontent.com/38169821/90331421-7325e180-dff7-11ea-9ccc-5ef6ae964d0b.PNG)

As far as I am aware, the lower the total loss score gives better mAP score.  If thats the case is it normal that the training yield the result shown? And also, is there any tips on increasing the accuracy of the model?

Any help would be greatly appreciated. Thanks

**Comment by abhigoku10** [comment]
@emiltan97  since you have around 9.5k images let the model train for at least 2lakh iterations, then you will start seeing some improvements in mAP on validation set, what is the initial loss it started during training ??

**Comment by emiltan97** [comment]
@abhigoku10  Hi. So my initial loss is : 
- "B": 5.22809, 
- "M": 4.35467, 
- "C": 15.33681, 
- "S": 3.54971, 
- "T": 28.46927

After training for about 40 hours (100,000 epochs) the loss decreases to : 
- "B": 0.61414, 
- "M": 2.37537, 
- "C": 1.69653, 
- "S": 0.43592, 
- "T": 5.12196

Unfortunately I cant find my final mAP score in the log file, if I remember correctly the result is not too much different from the image shown in my first comment. 

Slipping in an extra question : When labelling the dataset, are we supposed to give different colors to each instance of the same class? For example, in the images attached below, are each instances supposed to have different color?  
![41](https://user-images.githubusercontent.com/38169821/90485113-6e3d6b00-e17a-11ea-810a-3466780df553.jpg)
![41](https://user-images.githubusercontent.com/38169821/90485088-67aef380-e17a-11ea-8180-4637d6939d88.png)

Thanks, 
Emil.

**Comment by suchiz** [comment]
Hi there, I don't if you're still stuck here. But in your first screen, only 12k iterations has been done before calculating mAP. That is really poor. That's why your mAP is low. dbolya said you should wait at least 80k iterations to SEE something ... So 80k should probably have a low mAP.

Concerning the labelling. No, the mask is always binary. it shouldnt contain color. Only class number + binary mask

**Comment by emiltan97** [comment]
@suchiz  Thanks for the tips. I will train for 80k iterations next time and will close the issue for now.

**Comment by areebsyed** [comment]
@emiltan97 @suchiz  hello sorry for stupid question. I am training custom dataset and its showing me ETA 6,7 days. Is it the same time for you too? I want to reduce iterations. Where do I do it?

**Comment by suchiz** [comment]
Hello, you can change it in the data/config.py look for the variable max_iter

**Comment by ynma-hanvo** [comment]
@abhigoku10 how many training images do i need to prepare?  
if I only have 200 images ,  can i get a good model with 800000 iterations ?

**Comment by takwaaa** [comment]
Hi, 
Can someone please explain what do these values refer to ? thanks 
![image](https://user-images.githubusercontent.com/51539404/175790278-ad2a010e-8db5-41d7-a11a-d6554c1540ef.png)

**Comment by SdiZakariae** [comment]
> Hi, Can someone please explain what do these values refer to ? thanks ![image](https://user-images.githubusercontent.com/51539404/175790278-ad2a010e-8db5-41d7-a11a-d6554c1540ef.png)

hi takwaa  did you find the answer

**Comment by suchiz** [comment]
Hi @takwaaa @SdiZakariae , sorry for the late response.
These refer to the IoU used to calculate the mAP
